apps/emqx/src/emqx_trace/emqx_trace_json_formatter.erl

```diff
@@ -31,9 +33,10 @@
     LogEvent :: logger:log_event(),
     Config :: config().
 format(
-    LogMap0,
-    #{payload_encode := PEncode}
+    LogMap,
+    #{payload_encode := PEncode} = Config
 ) ->
+    LogMap0 = maybe_format_msg(LogMap, Config),
     LogMap1 = emqx_trace_formatter:evaluate_lazy_values(LogMap0),
```

Comment:
this should be done for other log formatters.
if you enable debug level logging, you'll notice that the anonymous function is logged.
e.g.
```
reason: {emqx_trace_format_func_data,#Fun<emqx_rule_runtime.1.32690002>,{{bridge_v2,http,cc},{error,{unrecoverable_error,#{headers => [{<<“location”>>,...
```

Reply:
then there should be no special handing in this formatter hence we can delegate the format/2 callback to emqx_logger_jsonfmt.erl ?

Reply:
Yes, created an issue for this so I think it can be done by another PR: https://emqx.atlassian.net/browse/EMQX-12380

Reply:
Actually, I think there is still some special handling. For example: restructuring error tuples into maps.